nixos/modules/services/monitoring/grafana-to-ntfy.nix

```diff
@@ -0,0 +1,125 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.services.grafana-to-ntfy;
+in
+{
+  options = {
+    services.grafana-to-ntfy = {
+      enable = lib.mkEnableOption "Grafana-to-ntfy (ntfy.sh) alerts channel";
+
+      package = lib.mkPackageOption pkgs "grafana-to-ntfy" { };
+
+      settings = {
+        ntfyUrl = lib.mkOption {
+          type = lib.types.str;
+          description = "The URL to the ntfy-sh topic.";
+          example = "https://push.example.com/grafana";
+        };
+
+        ntfyBAuthUser = lib.mkOption {
+          type = lib.types.nullOr lib.types.str;
+          description = ''
+            The ntfy-sh user to use for authenticating with the ntfy-sh instance.
+            Setting this option is required when using a ntfy-sh instance with access control enabled.
+          '';
+          default = null;
+          example = "grafana";
+        };
+
+        ntfyBAuthPass = lib.mkOption {
+          type = lib.types.path;
+          description = ''
+            The path to the password for the specified ntfy-sh user.
+            Setting this option is required when using a ntfy-sh instance with access control enabled.
+          '';
+          default = null;
+        };
+
+        bauthUser = lib.mkOption {
+          type = lib.types.str;
+          description = ''
+            The user that you will authenticate with in the Grafana webhook settings.
+            You can set this to whatever you like, as this is not the same as the ntfy-sh user.
+          '';
+          default = "admin";
+        };
+
+        bauthPass = lib.mkOption {
+          type = lib.types.path;
+          description = "The path to the password you will use in the Grafana webhook settings.";
+        };
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.services.grafana-to-ntfy = {
+      wantedBy = [ "multi-user.target" ];
+      wants = [ "network-online.target" ];
+      after = [ "network-online.target" ];
```

Comment:
```suggestion
```

Quoting https://systemd.io/NETWORK_ONLINE/:

> It is strongly recommended not to make use of this target too liberally: for example network server software should generally not pull this in (since server software generally is happy to accept local connections even before any routable network interface is up). Its primary purpose is network client software that cannot operate without network.

IMHO we should move away from `network-online.target` if the software is able to deal with flaky network conditions. I verified that `grafana-to-ntfy` will send [an error http status to grafana/alertmanager if the ntfy server can't be reached](https://github.com/kittyandrew/grafana-to-ntfy/blob/master/src/main.rs#L57-L69). Thus we can safely remove the dependencies.